### ISSUE TITLE
Refresh of Runtime - > Status causes increase of used heap size

### ISSUE THREAD

**Issue by rsvoboda** [hal/hal.next#129]
Refresh of Runtime - > Status causes increase of used heap size
I'm hitting Refresh "button" in right top corner of Status page, see screenshot attached to #128 

Using hal-standalone-0.9.1-20171009.081751-4.jar and WildFly 11.0.0.CR1
Windows 10, Java version "1.8.0_101"

In average 2 clicks on Refresh "button" are causing increase of 1 MB, it feels like too much to get update about uptime, heap memory and thread.

I see 2 options :) HAL is not efficiently using resources on server side or server itself creates a lot of objects to prepare the response.

When I clicked more than 100 times things get cleaned by GC,  from my perspective it's unnecessary garbage that must be cleaned. Spending time in GC just because of server interaction with web console is waste of resources.

**Comment by rsvoboda** [comment]
Did some experiments just with CLI and seems it's really server issue, still I would like to hear from @hpehl.

I was checking server via jconsole, 3 different approaches to get memory info data

**A) naive 100 loops of CLI invocation**
```
FOR /L %x IN (1,1,100) DO jboss-cli.bat -c "ls core-service=platform-mbean/type=memory"
```
This has huge overhead of establishing the connection.
Heap  went roughly 40 MB -> 170 MB -> 35 MB -> 140 MB ... allocation of 230 MB + 1x GC
It's interesting to see the price of 100 invocation.

**B) CLI file with 100 lines, using `ls` command**
```
FOR /L %x IN (1,1,100) DO echo ls core-service=platform-mbean/type=memory >> memory-info.cli
jboss-cli.bat -c --file=memory-info.cli
```
I tried to run the .cli file multiple times to have more valid data, memory increase was 39-40 MB

**C) CLI file with 100 lines, using `:read-resource`**
```
FOR /L %x IN (1,1,100) DO echo /core-service=platform-mbean/type=memory:read-resource(recursive=true, include-runtime=true) >> memory-info-read-resource.cli
jboss-cli.bat -c --file=memory-info-read-resource.cli
```
Did also multiple runs,  memory increase was 23-24 MB

**Comment by hpehl** [comment]
I did another test using the current HAL console showing the same results. Going to close this as it's not related to the console, but a server side issue.

**Comment by rsvoboda** [comment]
I did additional experiments and took heap dump before and after cli commands invocation.

tl;dr version: no issue confirmed

long story:
```bash
for i in {1..100}; do echo ls core-service=platform-mbean/type=memory >> memory-info.cli; done

PID=`jps -l | grep jboss-modules.jar | cut -d" " -f 1`
TODAY=`date +%Y-%m-%d.%H:%M:%S`

jcmd ${PID} GC.run && sleep 3
jcmd ${PID} GC.heap_dump server-${TODAY}-before-100-commands.hprof
jboss-eap-7.1/bin/jboss-cli.sh -c --file=memory-info.cli && sleep 1
jcmd ${PID} GC.heap_dump server-${TODAY}-after-100-commands.hprof
```
I used https://www.eclipse.org/mat/ to compare the dumps and nothing stood out, used size was almost the same

So I tried it again with jconsole running to monitor jvm in realtime, heap size goes down when JVM is asked for heap dump, seems temporarily used memory goes away. I tried to get the heap dump via jcmd and via jvisualvm UI tool, wait longer before heap dump. Heap size always went down when I asked for heap dump :)

Final check was to invoke 100 line cli script in 150 iterations, nothing came out from it. 
```bash
for i in {1..150}; do jboss-eap-7.1/bin/jboss-cli.sh -c --file=memory-info.cli 1>/dev/null; sleep 1;  done
```
